model/driver/src/icon4py/model/driver/icon4py_driver.py

```diff
@@ -114,23 +105,17 @@ def n_time_steps(self):
     def substep_timestep(self):
         return self._substep_timestep
 
-    def _swap(self):
-        time_n_swap = self._next
-        self._next = self._now
-        self._now = time_n_swap
-
     def _full_name(self, func: Callable):
         return ":".join((self.__class__.__name__, func.__name__))
 
     def time_integration(
         self,
         diffusion_diagnostic_state: diffusion_states.DiffusionDiagnosticState,
         solve_nonhydro_diagnostic_state: solve_nh_states.DiagnosticStateNonHydro,
-        # TODO (Chia Rui): expand the PrognosticState to include indices of now and next, now it is always assumed that now = 0, next = 1 at the beginning
-        prognostic_state_list: list[prognostics.PrognosticState],
+        prognostic_state_swp: imc_utils.Swapping[prognostics.PrognosticState],
```

Comment:
can we simply call it `prognostic_states` in plural? `prognostic_states.swap()` is kind of nice....